FuncGodot Manual/pages/tips_runtime_building.html

```diff
@@ -0,0 +1,123 @@
+
+    <h1>Building from Map Editors</h1>
+    <p>
+        Much like idTech engine games, it's possible to build and launch directly into a map from some map editors.
+        This can be quite a helpful tool for testing rapid iterations.
+    </p>
+
+    <h2>Godot Project Setup Example</h2>
+    <p>To build and launch our maps from the editor, you'll need to employ the <a href="https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-method-get-cmdline-user-args" target="_blank">OS class' ability to retrieve command line arguments</a>. A basic example might look something like this:</p>
+
+
+    <code><pre># This script should be autoloaded to handle command line arguments
+# Additionally, requires a valid reference to a FuncGodot map
+
+var func_godot_map: FuncGodotMap
+
+var map_override: String
+var cmd_line_args := OS.get_cmdline_user_args()
+
+# Loop over all command line arguments, split on a delimiter for each command, and use this for the file path
+for cmd_line_arg in cmd_line_args:
+    if cmd_line_arg.contains("run_map::"):
+        var arg_and_value := cmd_line_arg.split("::")
+        # This should split into an array of size 2, containing the parameter name and its corresponding value
+        if arg_and_value.size() != 2:
+            return
+
+        map_override = arg_and_value[1]
+
+        func_godot_map.local_map_file = map_override
+        func_godot_map.verify_and_build()</code></pre>
+    <h2>Setting Up Your Editor</h2>
+    <p>In the below guides, there will be references made to executables. This is up to the user, however you can use the Godot executable to run your game straight from the editor, or you could use exported binaries of your game. On Windows, your Godot Binary path is where you keep your Godot.exe file. On most Linux systems this is found under <code>/usr/bin/godot</code>. An example executable path and parameters for running a game in the editor: <code class="highlight">C:/GAMEDEV/Godot.exe --path C:/func_godot_project_example/ -- run_map::${MAP_BASE_NAME}.map</code>. Note that <code class="highlight">--path</code> is to the directory where project.godot resides. See the full list of Godot engine <a href="https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html" target="_blank">Command Line Interface options</a>.</p>
+    <h3>TrenchBroom</h3>
+    <p>
+        The TrenchBroom documenation <a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines">covers this generally</a>.<br>
+
+        To run your map from TrenchBroom, navigate to <code class="highlight">Run > Launch Engine</code>.
+        Go to <code class="highlight">Configure engines...</code>, add a game engine profile, and provide a name and path to the executable for this. For this example, we use the path to the Godot editor executable.
+        Once you're happy with your engine configuration, click close. You will now see this as an option you can select. Select it.
```

Comment:
Try to avoid using <br> breaks when a new <p> paragraph would be more fitting. Also, in keeping with the spirit of "Don't explain what the editor's manual already does better", figured this could use a rewrite.

```html
<p><a href="https://trenchbroom.github.io/manual/latest/#launching_game_engines"><b>The TrenchBroom Reference Manual covers launch option configurations quite thoroughly</b> If you chose to use the Godot Editor for your launch executable, make sure to start your parameters with the <code>--path</code> command. You'll then add the <code>--run_map::</code> build command we defined earlier, using the built-in TrenchBroom variable <code>${MAP_FULL_NAME}</code> to get the name of the currently opened map file.</a>.</p>
<p>An example of set of parameters to launch from the Godot Editor will look something like this: <code>--path "C:\GAME DEV\FuncGodot\test_project" --run_map::${MAP_FULL_NAME}</code></p>
```

Reply:
I've just tested, `${MAP_FULL_NAME}` doesn't work, and also the launch options needs the `--` as a separator(which requires a space around it on the left and right side otherwise it's seen as another Godot built in CLI argument which doesn't work).

`${MAP_FULL_NAME}` is available elsewhere in Trenchbroom as a variable but not for launching maps. Let me know if you're able to have any success with that?